### PR TITLE
#448 inbound line edit 9

### PR DESCRIPTION
**Pull request by joshxg**
The never ending #448 !

This is enabling saving/persisting of inbound updates and inserts. Haven't got to deletes yet.

Is working for mocks and the dev branch (after recreating the database again... the current dev branch references `location`, so if thats not in your schema 💥

### CODE REVIEW

File: packages/common/src/schema.graphql [hunk 1]
```diff
@@ -1082,6 +1082,15 @@ input BatchOutboundShipmentInput {
...
 
+input BatchInboundShipmentInput {
+  insertOutboundShipments: [InsertInboundShipmentInput!]
+  insertInboundShipmentLines: [InsertInboundShipmentLineInput!]
+  updateInboundShipmentLines: [UpdateInboundShipmentLineInput!]
+  deleteInboundShipmentLines: [DeleteInboundShipmentLineInput!]
+  updateInboundShipments: [UpdateInboundShipmentInput!]
+  deleteInboundShipments: [String!]
+}
+
```
Comment (joshxg) [inline packages/common/src/schema.graphql:1093]: This is for auto generating the type - maybe I should remove but it doesn't make any difference for the api 🤔

File: packages/common/src/utils/formatters.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 export const formatDate = (date: Date): string =>
   `${date.getDate()}/${date.getMonth() + 1}/${date.getFullYear()}`;
 
+export const formatNaiveDate = (date: Date): string => {
+  return formatDate(date).replace(/\//g, '-');
+};
```
Comment (joshxg) [inline packages/common/src/utils/formatters.ts:6]: 🙄

File: packages/invoices/src/InboundShipment/DetailView/api.ts [hunk 3]
```diff
@@ -87,21 +88,24 @@ const invoiceToInput = (
...
-    id: line.id,
-    itemId: line.itemId,
-    batch: line.batch,
-    costPricePerPack: line.costPricePerPack,
-    expiryDate: line.expiryDate,
-    sellPricePerPack: line.sellPricePerPack,
-    packSize: line.packSize,
-    numberOfPacks: line.numberOfPacks,
-    invoiceId: line.invoiceId,
+const createInsertInboundLineInput =
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/DetailView/api.ts:91]: just a few changes needed here to work with the backend which was nice!

File: packages/mock-server/src/api/mutations.ts [hunk 4]
```diff
@@ -44,9 +45,20 @@ export const insert = {
 
     return { ...createdInvoice, __typename: 'InvoiceNode' };
   },
-  invoiceLine: (
-    invoiceLine: InsertOutboundShipmentLineInput
-  ): InsertOutboundShipmentLineInput => {
+
```
Comment (joshxg) [inline packages/mock-server/src/api/mutations.ts:48]: whole mock server /worker are getting pretty messy. Not sure what the plan is - I suppose keeping it around for awhile longer so might be worth a couple hour cleanup pass soon 🤔

**Review by joshxg — COMMENTED** [review]
Mainly updating the mock server - everything else seemed to just work pretty well

File: packages/mock-server/src/data/database.ts [hunk 5]
```diff
@@ -164,25 +165,49 @@ export const insert = {
...
       itemCode: item.code,
       itemUnit: item.unitName ?? '',
       itemId: item.id,
-      expiryDate: stockLine.expiryDate,
-      stockLineId: stockLine.id,
-      costPricePerPack: stockLine.costPricePerPack,
-      sellPricePerPack: stockLine.sellPricePerPack,
-      location: stockLine.location,
-      packSize: stockLine.packSize,
+      expiryDate: 'stockLine?.expiryDate',
```
Comment (mark-prins) [inline packages/mock-server/src/data/database.ts:201]: are the quotes needed now?

**Review by mark-prins — APPROVED** [review]
🎉  😮  wow! and again.. nice work. The endless 448 is going down.. eventually!!

File: packages/mock-server/src/data/database.ts (hunk 5 above)
Comment (joshxg) [reply packages/mock-server/src/data/database.ts:201]: ah thanks. Got it in the next one!

**Pull request merged**